src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -95,18 +95,14 @@ typedef enum {
     /* 6 */ ZELDA_EYE_STATE_CLOSED2
 } DmZlEyeStates;
 
-/**
- * This function is always called with unusedExtraOffset = 0.
- */
-void DmZl_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo animationInfo[], u16 unusedExtraOffset) {
+void DmZl_ChangeAnim(SkelAnime* skelAnime, AnimationInfo* animInfo, u16 animIndex) {
     f32 endFrame;
 
-    animationInfo += unusedExtraOffset;
-    endFrame = (animationInfo->frameCount < 0.0f) ? Animation_GetLastFrame(animationInfo->animation)
-                                                  : animationInfo->frameCount;
+    animInfo += animIndex;
+    endFrame = (animInfo->frameCount < 0.0f) ? Animation_GetLastFrame(animInfo->animation) : animInfo->frameCount;
```

Comment:
slightly different here. Still with the `endframe` temp, but uses a ternary to set it.